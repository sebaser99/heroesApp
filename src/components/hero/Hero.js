import { useMemo } from "react";
import { Navigate, useNavigate, useParams } from "react-router-dom";
import { getHeroById } from "../../selectors/getHeroById";


export const Hero = () => {
    const {heroId} = useParams()
    const hero = useMemo(()=> getHeroById(heroId), [heroId]); 
    const navigate = useNavigate()
    
    if(!hero){
        return <Navigate to='/' />
    }
    const {id, superhero, characters, alter_ego, publisher, first_appearance } = hero;

    const handleClick = ()=>{
        // if(publisher === 'DC Comics'){
        //     navigate('/dc')
        // }else{
        //     navigate('/marvel')
        // }
        navigate(-1)
    }

    const imagePath = `/statics/images/${id}.jpg`
    return (
        <div className="row mt-5">
            <div className="col-4">
                <img 
                    src={imagePath}
                    alt={superhero}
                    className="img-thumbnail animate__animated animate__bounceInLeft"
                />
            </div>
            <div className="col-8 animate__animated animate__fadeIn">
                <h3>{superhero}</h3>
                <ul className="list-group list-group-flush">
                    <li className="list-group-item"><b>Alter Ego:</b> {alter_ego}</li>
                    <li className="list-group-item"><b>Publisher:</b> {publisher}</li>
                    <li className="list-group-item"><b>First Appearance:</b> {first_appearance}</li>
                </ul>

                <h5 className="mt-5">Characters</h5>
                <p>{characters}</p>

                <button
                    className="btn btn-outline-info"
                    onClick={handleClick}
                >Regresar</button>

            </div>
            
        </div>
    )
}
