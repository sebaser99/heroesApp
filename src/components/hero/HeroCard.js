import { Link } from "react-router-dom";
import { heroesImages } from "../../helpers/heroImages";

export const HeroCard = ({
    id,
    superhero,
    publisher,
    alter_ego,
    first_appearance,
    characters,
}) => {
    // const imagePath = `/statics/images/${id}.jpg`
    return (
        <div className="col" >
            <div className="card" style={{minHeight: 290}}>
                <div className="row no-gutters">
                <div className="col-4 ">
                    <img className="card-img-top" 
                        // src={imagePath} 
                        src={heroesImages(`./${id}.jpg`)}
                        alt={superhero} />
                </div>
                <div className="col-8">
                    <div className="card-body">
                        <h5 className="card-title">{superhero}</h5>
                        <p className="card-text">{alter_ego}</p>
                        {(alter_ego !== characters) && 
                            <p >{characters}</p>
                        }
                        <p className="card-text">
                            <small >{first_appearance}</small>
                        </p>
                        <Link className="btn btn-outline-warning"  to={`/hero/${id }`}>
                            Ver mÃ¡s ...
                        </Link>
                    </div>

                </div>
                </div>
                
            </div>
        </div>
    )
}
