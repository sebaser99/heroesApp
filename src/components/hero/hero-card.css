.card {
    background-color: #504a4a;
    color: white;
}
.text-muted .white {
    color: white;
}