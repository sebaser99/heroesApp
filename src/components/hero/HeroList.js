import { useMemo } from 'react';
import { getHeroesByPublisher } from '../../selectors/getHeroesByPublisher';
import { HeroCard } from './HeroCard';

export const HeroList = ({publisher}) => {
    const heroes = useMemo(()=> getHeroesByPublisher(publisher), [publisher]) 
    return (
        <div className='row rows-col-1 row-cols-1 row-cols-sm-2  row-cols-md-3 g-3 animate__animated animate__fadeIn'>

            {
                heroes.map(hero =>(
                    <HeroCard key={hero.id}
                        {...hero}
                   />
                ))
            }
        </div>
    )
}
