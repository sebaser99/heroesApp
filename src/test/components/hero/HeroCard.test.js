import { shallow } from "enzyme";
import { HeroCard } from "../../../components/hero/HeroCard";

describe('vamos a probar <HeroCard />', ()=>{
    test('debe hacer match con el snapshot', ()=>{
        const hero = {
            'id': 'dc-batman',
            'superhero':'Batman', 
            'publisher':'DC Comics', 
            'alter_ego':'Bruce Wayne',
            'first_appearance':'Detective Comics #27',
            'characters':'Bruce Wayne'
        }
        
        const wrapper = shallow(<HeroCard  id={hero.id}
            superhero={hero.superhero} publisher={hero.publisher}
            alter_ego={hero.alter_ego} first_appearance={hero.first_appearance}
            characters={hero.characters}
        />)
        expect(wrapper).toMatchSnapshot()
    })
    
})