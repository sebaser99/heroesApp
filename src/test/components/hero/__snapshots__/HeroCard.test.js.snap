// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vamos a probar <HeroCard /> debe hacer match con el snapshot 1`] = `
<div
  className="col"
>
  <div
    className="card"
    style={
      Object {
        "height": 300,
      }
    }
  >
    <div
      className="row no-gutters"
    >
      <div
        className="col-4 "
      >
        <img
          alt="Batman"
          className="card-img-top"
          src="/statics/images/dc-batman.jpg"
        />
      </div>
      <div
        className="col-8"
      >
        <div
          className="card-body"
        >
          <h5
            className="card-title"
          >
            Batman
          </h5>
          <p
            className="card-text"
          >
            Bruce Wayne
          </p>
          <p
            className="card-text"
          >
            <small
              className="text-muted"
            >
              Detective Comics #27
            </small>
          </p>
          <Link
            to="/hero/dc-batman"
          >
            Ver m√°s ...
          </Link>
        </div>
      </div>
    </div>
  </div>
</div>
`;
